web/src/settings_account.ts

```diff
@@ -675,6 +678,50 @@ export function set_up(): void {
         );
     }
 
+    function render_user_timezone_dropdown_widget(selected_timezone?: string): void {
+        const timezone_items = timezones.timezones.map(
+            (tz: {name: string; utc_offset: string}) => ({
+                name: `${tz.name} (${tz.utc_offset})`,
+                unique_id: tz.name,
+            }),
+        );
+
+        const opts: dropdown_widget.DropdownWidgetOptions = {
+            widget_name: "user_timezone",
+            get_options: () => timezone_items,
+            item_click_callback(event, dropdown, widget) {
+                dropdown.hide();
+                event.preventDefault();
+                event.stopPropagation();
+                const selected = widget.value();
+                settings_ui.do_settings_change(
+                    channel.patch,
+                    "/json/settings",
+                    {timezone: selected},
+                    $(".timezone-setting-status").expectOne(),
+                    {
+                        success_continuation() {
+                            /*
+                            When user selects the option, update the widget to show the selected option.
```

Comment:
Can we just select the option immediately and then reset to the original value in case of error? I think that would be better.